godot-macros/src/derive/derive_from_variant.rs

```diff
@@ -161,42 +164,42 @@ fn make_tuple_struct(
                 }
             } else {
                 quote! {
-                    let #ident = root.pop_front()?
-                                     .try_to::<#field_type>()
-                                     .ok()?;
+                    let #ident = root.pop_front()
+                                     .ok_or(ConvertError::custom().with_cause("missing expected value").with_value(root.clone()))?
+                                     .try_to::<#field_type>()?;
```

Comment:
I would do indentation like regular rustfmt: from start of the scope, not aligned with the first `.`